ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/answers/activity-file-answer/activityFileAnswer.component.ts

```diff
@@ -1,51 +1,74 @@
 import {
     Component,
+    ElementRef,
     EventEmitter,
     Input,
     OnDestroy,
-    Output
+    OnInit,
+    Output,
+    ViewChild
 } from '@angular/core';
+import { MatDialog } from '@angular/material/dialog';
 
 import { EMPTY, Subject } from 'rxjs';
 import { catchError, takeUntil } from 'rxjs/operators';
 
-import { FileUploadBody, FileUploadResponse } from '../../../../models/fileUpload';
+import { UploadFile } from '../../../../models/uploadFile';
 import { ActivityFileQuestionBlock } from '../../../../models/activity/activityFileQuestionBlock';
 import { FileUploadService } from '../../../../services/fileUpload.service';
 import { LoggingService } from '../../../../services/logging.service';
+import { ActivityFileAnswerDto } from '../../../../models/activity/activityFileAnswerDto';
+import { FileUploadBody } from '../../../../models/fileUploadBody';
+import { FileUploadResponse } from '../../../../models/fileUploadResponse';
+import { ModalService } from '../../../../services/modal.service';
+import { ConfirmDialogComponent } from '../../../confirmDialog/confirmDialog.component';
+import { ActivityFileValidationRule } from '../../../../services/activity/validators/activityFileValidationRule';
 
 @Component({
     selector: 'ddp-activity-file-answer',
     templateUrl: 'activityFileAnswer.component.html',
     styleUrls: ['activityFileAnswer.component.scss']
 })
-export class ActivityFileAnswer implements OnDestroy {
+export class ActivityFileAnswer implements OnInit, OnDestroy {
     @Input() block: ActivityFileQuestionBlock;
     @Input() readonly: boolean;
     @Input() studyGuid: string;
     @Input() activityGuid: string;
     @Output() valueChanged: EventEmitter<string | null> = new EventEmitter();
+    @ViewChild('uploadBtn', {read: ElementRef}) private uploadButtonRef: ElementRef;
+    @ViewChild('undoUploadBtn', {read: ElementRef}) private undoUploadButtonRef: ElementRef;
 
-    file: File;
-    fileUploadData: FileUploadResponse;
+    selectedFile: File;
     isFileSelected: boolean;
+    uploadedFile: UploadFile | null;
+    fileMaxSize: number;
+    fileMimeTypes: string[];
+    errorMessage: string;
+    private fileToUpload: UploadFile | null;
```

Comment:
it seems like it was better before with fileUploadData property. In fact, fileToUpload property is a combination of fileUploadData property and selectedFile property which means we duplicate information in the class instance state. I'd get rid of selectedFile then or get back to fileUploadData and just create the fileToUpload object locally in methods where it's needed.

Reply:
Refactored at 497c9ebd1345abb708c9fd0851e40d38fe7405b5